src/modules/home/components/Trending.tsx

```diff
@@ -0,0 +1,49 @@
+import { useQuery } from "@tanstack/react-query";
+import { ITrendingMangaApp } from "../../common/types/app-types";
+import { ITrendingManga } from "../../common/types/subtypes";
+import useMangaStore from '../../common/stores/store'
+import { fetchTrendingMangas } from "../../common/services/api-service";
+import TrendingList from "./TrendingList";
+import TrendingSkeleton from "./TrendingSkeleton";
+
+function Trending() {
+
+    const { isPending, error, data: topComicsResponse } = useQuery({
+        queryKey: ['fetchTrendingMangas'],
+        queryFn: fetchTrendingMangas
+    })
+
+    if (isPending) {
+        return <TrendingSkeleton />
+    }
+
+    if (error) {
+        return <div>Error: {error.message}</div>
+    }
+
+    if (topComicsResponse) {
+        const trendingMangas: ITrendingMangaApp[] = topComicsResponse.trending[30].map((manga: ITrendingManga) => {
+            return {
+                slug: manga.slug,
+                title: manga.title!,
+                coverImage: useMangaStore.getState().imageUrlPrefix + manga.md_covers[0].b2key,
+            }
+
+        })
+        useMangaStore.setState({ trendingMangas: trendingMangas })
+    }
```

Comment:
Perhaps you could add some error checks here before using the response? For example, checking if trending, slug, title and `md_covers?.[0]?.b2key` exists?